languages/scala/reference/exercise-concepts/hamming.md

```diff
@@ -0,0 +1,32 @@
+# Concepts of hamming
+
+[Example implementation](https://github.com/exercism/scala/blob/master/exercises/hamming/example.scala)
+
+## General
+
+- object or case class: used as class wrapper for exercies
+- functions: used as main entry point for the exercise
+- function arguments: input strands passed as arguments
+- return values: return a value from a function
+- conditionals using if: conditionally execute logic using an `if` statement
+- strings: `string` passed as the two input parameters
+- character operations: `!=`
+- integer: `Int` is used as return result within `Option`
+- Option: `Option` is used as return result
+- math operations: `==`
+- zip: combine two strings into a sequence of tuples
+- count: count number of lambda calls that return true
+- lambda: define lambda to compare characters
+
+## Using `filter` and `size` to count differences
+
+- filter: remove matching characters from zipped sequence
+- lambda: define lambda to compare characters
+- size: calculate size of sequence
```

Comment:
Similar comment to an earlier comment on implementation vs. general concept.

Reply:
Maybe - "collections - length"?

Reply:
Ehm, isn't this a special case of "collections - reduce"? I.e. having a collection and reducing it to a single value?

Reply:
I am not so sure. I see your point. But length/size seems like simple/native functionality on a collection. Where reduce seems like more advanced. A student could easily understand what length means without having any idea how to reduce a collection.

Reply:
I see. But is the Concept then maybe simply method types? The thing I'm getting at, is that there is not much special to the length of a collection in terms of Concepts.

Reply:
I see - then maybe this falls under a general "collections" concept? As opposed to "collections - reduce". I still think "reduce" is a different concept.

Reply:
I could have changed this @ErikSchierboom  - you didn't have to merge with `collections-length` in there.. If you still want me to change it to something else, it is not a problem.

Reply:
No worries, we can look at this later.